sites/svelte-5-preview/src/routes/docs/content/01-api/02-runes.md

```diff
@@ -159,48 +159,45 @@ In essence, `$derived(expression)` is equivalent to `$derived.by(() => expressio
 
 ## `$effect`
 
-To run side-effects like logging or analytics whenever some specific values change, or when a component is mounted to the DOM, we can use the `$effect` rune:
+To run _side-effects_ when the component is mounted to the DOM, and when values change, we can use the `$effect` rune ([demo](/#H4sIAAAAAAAAE31T24rbMBD9lUG7kAQ2sbdlX7xOYNk_aB_rQhRpbAsU2UiTW0P-vbrYubSlYGzmzMzROTPymdVKo2PFjzMzfIusYB99z14YnfoQuD1qQh-7bmdFQEonrOppVZmKNBI49QthCc-OOOH0LZ-9jxnR6c7eUpOnuv6KeT5JFdcqbvbcBcgDz1jXKGg6ncFyBedYR6IzLrAZwiN5vtSxaJA-EzadfJEjKw11C6GR22-BLH8B_wxdByWpvUYtqqal2XB6RVkG1CoHB6U1WJzbnYFDiwb3aGEdDa3Bm1oH12sQLTcNPp7r56m_00mHocSG97_zd7ICUXonA5fwKbPbkE2ZtMJGGVkEdctzQi4QzSwr9prnFYNk5hpmqVuqPQjNnfOJoMF22lUsrq_UfIN6lfSVyvQ7grB3X2mjMZYO3XO9w-U5iLx42qg29md3BP_ni5P4gy9ikTBlHxjLzAtPDlyYZmRdjAbGq7HprEQ7p64v4LU_guu0kvAkhBim3nMplWl8FreQD-CW20aZR0wq12t-KqDWeBywhvexKC3memmDwlHAv9q4Vo2ZK8KtK0CgX7u9J8wXbzdKv-nRnfF_2baTqlYoWUF2h5efl9-n0O6koAMAAA==)):
 
 ```svelte
 <script>
-	let count = $state(0);
-	let doubled = $derived(count * 2);
+	let size = $state(50);
+	let color = $state('#ff3e00');
+
+	let canvas;
 
 	$effect(() => {
-		console.log({ count, doubled });
+		const context = canvas.getContext('2d');
+		context.clearRect(0, 0, canvas.width, canvas.height);
+
+		// this will re-run whenever `color` or `size` change
+		context.fillStyle = color;
+		context.fillRect(0, 0, size, size);
 	});
 </script>
 
-<button on:click={() => count++}>
-	{doubled}
-</button>
-
-<p>{count} doubled is {doubled}</p>
+<canvas bind:this={canvas} width="100" height="100" />
 ```
 
-`$effect` will automatically subscribe to any `$state` or `$derived` values it reads _synchronously_ and reruns whenever their values change — that means, values after an `await` or inside a `setTimeout` will _not_ be tracked. `$effect` will run after the DOM has been updated.
+The function passed to `$effect` will run when the component mounts, and again — after any updates to the DOM — whenever there are changes to any values it reads that were declared with `$state` or `$derived` (including those passed in with `$props`).
```

Comment:
This reads a bit crammed with the `- after any .. -`. Not sure yet how to best model this instead, maybe split it up into two sentences?

```suggestion
The function passed to `$effect` will run when the component mounts, and again whenever there are changes to any values it reads that were declared with `$state` or `$derived` (including those passed in with `$props`). The timing of reruns is after updates to the DOM.
```